tests/units/fastsync/commons/test_split_gzip.py

```diff
@@ -0,0 +1,149 @@
+        Test if passing invalid parameters raising exceptions
+        """
+        with self.assertRaises(ValueError):
+            split_gzip.open('basefile', mode='invalidmode')
+        with self.assertRaises(ValueError):
+            split_gzip.open('basefile', mode='wt', chunk_size_mb=0)
+        with self.assertRaises(ValueError):
+            split_gzip.open('basefile', max_chunks=-1)
+
+    # pylint: disable=W0212
+    def test_gen_export_chunk_filename(self):
+        """
+        Test generating chunked filenames
+        """
+        # split_large_files should be disabled when max_chunks is zero
+        gzip_splitter = split_gzip.SplitGzipFile('basefile', chunk_size_mb=1000, max_chunks=0)
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile')
+        # first chunk should be part nr 1
+        gzip_splitter = split_gzip.SplitGzipFile('basefile', chunk_size_mb=1000, max_chunks=20)
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00001')
+        # generated file part should be in sync with chunk_seq
+        gzip_splitter.chunk_seq = 5
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00005')
+        # chunk seq should not increase if chunk_size is lower than split_file_chunk_size_mb
+        gzip_splitter.current_chunk_size_mb = 500
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00005')
+        # chunk seq should increase and size should reset if current_chunk_size_mb greater than chunk_size_mb
+        gzip_splitter.current_chunk_size_mb = 1050
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00006')
+        self.assertEqual(gzip_splitter.current_chunk_size_mb, 0)
+        # chunk seq should not increase further if chunk_seq equals to split_file_max_chunks
+        gzip_splitter.chunk_seq = 20
+        self.assertEqual(gzip_splitter._gen_chunk_filename(), 'basefile.part00020')
+
+    def test_write_with_no_split(self):
+        """
+        Write gzip without splitting it and reading it
+        """
+        # max_chunk = 0 should create a file with no splitting
+        with split_gzip.SplitGzipFile(self.filename, 'wb', max_chunks=0) as f_write:
+            f_write.write(DATA_WITH_100_BYTES * 50)
+
+        with gzip.open(self.filename, 'rb') as f_read:
+            file_content = f_read.read()
+
+        self.assertEqual(file_content, DATA_WITH_100_BYTES * 50)
+
+    def test_write_with_single_chunk(self):
+        """
+        Write all data into one chunk
+        """
+        # test data fits into one chunk
+        with split_gzip.SplitGzipFile(self.filename, 'wb', chunk_size_mb=1000, max_chunks=20) as f_write:
+            f_write.write(DATA_WITH_100_BYTES * 50)
+
+        with gzip.open(f'{self.filename}.part00001', 'rb') as f_read:
+            file_content = f_read.read()
+
+        self.assertEqual(file_content, DATA_WITH_100_BYTES * 50)
+
+    def test_write_with_multiple_chunks(self):
```

Comment:
Can you add a similar test where you reach the `max_chunks` but the last file size is greater than the `chunk_size_mb`?

Reply:
added by https://github.com/transferwise/pipelinewise/pull/696/commits/76cbd0acd414ae95b259b331a1e6d6ca58826814